Realm/Realm.SourceGenerator/Analytics.cs

```diff
@@ -0,0 +1,524 @@
+            { SdkFeature.Facebook, 0 },
+            { SdkFeature.Google, 0 },
+            { SdkFeature.Apple, 0 },
+            { SdkFeature.JWT, 0 },
+            { SdkFeature.ApiKey, 0 },
+            { SdkFeature.ServerApiKey, 0 },
+            { SdkFeature.Function, 0 },
+            { SdkFeature.CallAsync, 0 },
+            { SdkFeature.FIX_ME_THREE, 0 },
+            { SdkFeature.DynamicApi, 0 },
+        };
+
+        private Dictionary<string, string> _realmEnvMetrics = new()
+        {
+            { SdkFeature.UserId, string.Empty },
+            { SdkFeature.RealmSdk, ".NET" },
+            { SdkFeature.Language, "C#" },
+            { SdkFeature.HostOsType, string.Empty },
+            { SdkFeature.HostOsVersion, string.Empty },
+        };
+
+        private static string AnonymizedUserID
+        {
+            get
+            {
+                try
+                {
+                    var id = GenerateComputerIdentifier();
+                    return id != null ? SHA256Hash(id) : "UNKNOWN";
+                }
+                catch
+                {
+                    return "UNKNOWN";
+                }
+            }
+        }
+
+        public void SubmitAnalytics(GeneratorExecutionContext context)
+        {
+            var payload = string.Empty;
+
+            if (Environment.GetEnvironmentVariable("REALM_DISABLE_ANALYTICS") != null ||
+                Environment.GetEnvironmentVariable("CI") != null)
+            {
+                return;
+            }
+
+            _realmEnvMetrics[SdkFeature.UserId] = AnonymizedUserID;
+
+            ComputeHostOSNameAndVersion(out var osName, out var osVersion);
+
+            // TODO andrea: check if searching strings in this way works on non Windows platforms
+            switch (osName)
+            {
+                case string name when name.Contains(Metric.OperatingSystem.Windows):
+                    _realmEnvMetrics[SdkFeature.HostOsType] = Metric.OperatingSystem.Windows;
+                    break;
+                case string name when name.Contains(Metric.OperatingSystem.MacOS):
+                    _realmEnvMetrics[SdkFeature.HostOsType] = Metric.OperatingSystem.MacOS;
+                    break;
+                case string name when name.Contains(Metric.OperatingSystem.Linux):
+                    _realmEnvMetrics[SdkFeature.HostOsType] = Metric.OperatingSystem.Linux;
+                    break;
+                default:
+                    Diagnostics.Add(
+                        SourceGenerator.Diagnostics.AnalyticsDebugInfo($"{osName} is not an operating system that we recognize."));
+                    break;
+            }
+
+            _realmEnvMetrics[SdkFeature.HostOsVersion] = osVersion;
+
+            bool prettyJson = false;
+#if DEBUG
+            prettyJson = true;
+#endif
+            payload = GetJsonPayload(prettyJson);
+
+#if !DEBUG
+            var base64Payload = Convert.ToBase64String(Encoding.UTF8.GetBytes(payload));
```

Comment:
This should be probably be part of the initial check of this method. If we're not going to send the analytics we shouldn't even collect them

Reply:
This is about SGs. We've now moved this to the weaver.